pkg/volume/csi/csi_plugin.go

```diff
@@ -246,16 +247,90 @@ func (p *csiPlugin) Init(host volume.VolumeHost) error {
 		}
 	}
 
-	// Initializing csiDrivers map and label management channels
+	// Initializing csiDrivers map, label management channels, and migratedDrivers map
 	csiDrivers = csiDriversStore{driversMap: map[string]csiDriver{}}
-	nim = nodeinfomanager.NewNodeInfoManager(host.GetNodeName(), host)
 
-	// TODO(#70514) Init CSINodeInfo object if the CRD exists and create Driver
-	// objects for migrated drivers.
+	var migratedDrivers = map[string](func() bool){
+		csiPlugins.GCEPDDriverName: func() bool {
+			return utilfeature.DefaultFeatureGate.Enabled(features.CSIMigration) && utilfeature.DefaultFeatureGate.Enabled(features.CSIMigrationGCE)
+		},
+		// TODO(leakingtpan): Add AWS migration feature gates and place them here
+	}
+
+	nim = nodeinfomanager.NewNodeInfoManager(host.GetNodeName(), host, migratedDrivers)
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.CSINodeInfo) &&
+		utilfeature.DefaultFeatureGate.Enabled(features.CSIMigration) {
+		// This function prevents Kubelet from posting Ready status until CSINodeInfo
+		// is both installed and initialized
+		initializeCSINodeInfo(host)
+	}
 
 	return nil
 }
 
+func initializeCSINodeInfo(host volume.VolumeHost) {
+	kubeClient := host.GetKubeClient()
+	if kubeClient == nil || len(host.GetPodsDir()) == 0 {
+		// If we're NOT on the Kubelet OR KubeClient DOESNT exist
+		klog.V(4).Infof("Skipping CSINodeInfo initialization, not running on Kubelet or KubeClient doesn't exist")
+		return
+	}
+
+	go func() {
+		host.SetKubeletError(fmt.Errorf("Could not find CSINodeInfo CRD. Please install the CSINodeInfo CRD"))
+
+		// TODO(dyzz) fine tune the backoff parameters for CRD Installation and Initialize migrated drivers
+		crdBackoff := wait.Backoff{
+			Steps:    4,
+			Duration: 10 * time.Millisecond,
+			Factor:   5.0,
+			Jitter:   0.1,
+		}
+		for {
```

Comment:
combining `for` + `wait.ExponentialBackoff` has an odd burst pattern, where the attempts backoff until they reach the max exponential steps, then exit the ExponentialBackoff with an error, then restart the for loop fast again.

might be clearer to do something like:
```go
backoff := &wait.Backoff{
  Steps: 5,
  Duration: 10 * time.Millisecond,
  Factor: 4.0,
  Jitter: 0.1,
  Cap: 10 * time.Second,
}
for {
  time.Sleep(backoff.Step())
  groupResource, err := kubeClient.Discovery().ServerResourcesForGroupVersion("csi.storage.k8s.io" + "/" + "v1alpha1")
  if err != nil {
    ... log, report, continue ...
  }
  ... look for resource, if missing, log, report, continue
  break
}
```

that lets the backoff grow and settle at 10 second interval checks

same comment applies to the for/ExponentialBackoff below. would need to reset the Backoff struct

Reply:
an eventual "give up, go out, and come back in again" restart is probably better. I'm not sure what the other init-time checks do w.r.t. specific timeframes. Would recommend asking @kubernetes/sig-node-pr-reviews about what other subsystems (network, container runtime, etc) do